diff --git a/device/motorola/eqe/BoardConfig.mk b/device/motorola/eqe/BoardConfig.mk
index 52b3251..1c00d36 100644
--- a/device/motorola/eqe/BoardConfig.mk
+++ b/device/motorola/eqe/BoardConfig.mk
@@ -2,7 +2,7 @@
 # SPDX-FileCopyrightText: 2022-2024 The LineageOS Project
 # SPDX-License-Identifier: Apache-2.0
 #
-
+DISABLE_ARTIFACT_PATH_REQUIREMENTS := true
 DEVICE_PATH := device/motorola/eqe
 
 # Architecture
diff --git a/device/motorola/eqe/lineage_eqe.mk b/device/motorola/eqe/lineage_eqe.mk
index 80018ba..d784b99 100644
--- a/device/motorola/eqe/lineage_eqe.mk
+++ b/device/motorola/eqe/lineage_eqe.mk
@@ -12,9 +12,15 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base_telephony.mk)
 # Inherit from eqe device
 $(call inherit-product, device/motorola/eqe/device.mk)
 
+TARGET_HAS_UDFPS := true
+TARGET_INCLUDE_ACCORD := false
+
 # Inherit some common Lineage stuff.
 $(call inherit-product, vendor/lineage/config/common_full_phone.mk)
 
+# Include signing keys
+include vendor/private/keys/keys.mk
+
 PRODUCT_NAME := lineage_eqe
 PRODUCT_DEVICE := eqe
 PRODUCT_MANUFACTURER := motorola
diff --git a/device/motorola/eqe/overlay-lineage/packages/apps/Updater/app/src/main/res/values/strings.xml b/device/motorola/eqe/overlay-lineage/packages/apps/Updater/app/src/main/res/values/strings.xml
new file mode 100644
index 0000000..4575c92
--- /dev/null
+++ b/device/motorola/eqe/overlay-lineage/packages/apps/Updater/app/src/main/res/values/strings.xml
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <string
+        name="updater_server_url"
+        translatable="false"
+    >https://raw.githubusercontent.com/SomeEmptyBox/android_build_utils/refs/heads/main/ota/eqe/EvolutionX.json</string>
+    <string
+        name="menu_changelog_url"
+        translatable="false"
+    >https://raw.githubusercontent.com/SomeEmptyBox/android_build_utils/refs/heads/main/ota/eqe/EvolutionX.txt</string>
+</resources>
diff --git a/device/motorola/eqe/device.mk b/device/motorola/eqe/device.mk
index 904890b..158b04b 100644
--- a/device/motorola/eqe/device.mk
+++ b/device/motorola/eqe/device.mk
@@ -341,12 +341,6 @@ PRODUCT_COPY_FILES += \
 PRODUCT_PACKAGES += \
     android.hardware.power-service-qti
 
-# PowerShare
-PRODUCT_PACKAGES += \
-    vendor.lineage.powershare-service.default
-
-$(call soong_config_set,lineage_powershare,powershare_path,/sys/class/power_supply/wireless/device/tx_mode)
-
 # Protobuf
 PRODUCT_PACKAGES += \
     libprotobuf-cpp-full-3.9.1-vendorcompat \
