diff --git a/AndroidProducts.mk b/AndroidProducts.mk
index da4585f..546d6f8 100644
--- a/device/motorola/eqe/AndroidProducts.mk
+++ b/device/motorola/eqe/AndroidProducts.mk
@@ -5,4 +5,4 @@
 #
 
 PRODUCT_MAKEFILES := \
-    $(LOCAL_DIR)/lineage_eqe.mk
+    $(LOCAL_DIR)/infinity_eqe.mk
diff --git a/BoardConfig.mk b/BoardConfig.mk
index 28cbd6c..32673a7 100644
--- a/device/motorola/eqe/BoardConfig.mk
+++ b/device/motorola/eqe/BoardConfig.mk
@@ -176,7 +176,7 @@ DEVICE_FRAMEWORK_MANIFEST_FILE += $(DEVICE_PATH)/framework_manifest.xml
 DEVICE_FRAMEWORK_COMPATIBILITY_MATRIX_FILE := \
     $(DEVICE_PATH)/device_framework_matrix.xml \
     hardware/qcom-caf/common/vendor_framework_compatibility_matrix.xml \
-    vendor/lineage/config/device_framework_matrix.xml
+    vendor/infinity/config/device_framework_matrix.xml
 DEVICE_MATRIX_FILE := hardware/qcom-caf/common/compatibility_matrix.xml
 DEVICE_MANIFEST_SKUS := crow
 DEVICE_MANIFEST_CROW_FILES += \
@@ -194,7 +194,7 @@ BOARD_USES_METADATA_PARTITION := true
 # Partitions
 TARGET_RO_FILE_SYSTEM_TYPE ?= ext4
 ifneq ($(TARGET_RO_FILE_SYSTEM_TYPE),erofs)
--include vendor/lineage/config/BoardConfigReservedSize.mk
+-include vendor/infinity/config/BoardConfigReservedSize.mk
 else
 BOARD_EROFS_COMPRESSOR := lz4
 BOARD_EROFS_PCLUSTER_SIZE := 262144
diff --git a/lineage_eqe.mk b/infinity_eqe.mk
similarity index 78%
rename from lineage_eqe.mk
rename to infinity_eqe.mk
index 51a60b3..013ec8b 100644
--- a/device/motorola/eqe/lineage_eqe.mk
+++ b/device/motorola/eqe/infinity_eqe.mk
@@ -12,8 +12,16 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base_telephony.mk)
 # Inherit from eqe device
 $(call inherit-product, device/motorola/eqe/device.mk)
 
+INFINITY_MAINTAINER := "Peace"
+TARGET_SUPPORTS_BLUR := true
+WITH_GAPPS := true
+TARGET_SHIPS_FULL_GAPPS := true
+
 # Inherit some common Lineage stuff.
-$(call inherit-product, vendor/lineage/config/common_full_phone.mk)
+$(call inherit-product, vendor/infinity/config/common_full_phone.mk)
+
+# Include signing keys
+include vendor/private/keys/keys.mk
 
 PRODUCT_NAME := lineage_eqe
 PRODUCT_DEVICE := eqe
diff --git a/livedisplay/Android.bp b/livedisplay/Android.bp
deleted file mode 100644
index 1fd3eca..0000000
--- a/device/motorola/eqe/livedisplay/Android.bp
+++ /dev/null
@@ -1,38 +0,0 @@
-//
-// SPDX-FileCopyrightText: 2019-2024 The LineageOS Project
-// SPDX-License-Identifier: Apache-2.0
-//
-
-cc_defaults {
-    name: "livedisplay_eqe",
-    defaults: ["hidl_defaults"],
-    relative_install_path: "hw",
-    srcs: [
-        ":vendor.lineage.livedisplay@2.0-sdm-pa",
-        ":vendor.lineage.livedisplay@2.0-sdm-utils",
-        "AntiFlicker.cpp",
-        "SunlightEnhancement.cpp",
-        "service.cpp",
-    ],
-    shared_libs: [
-        "libbase",
-        "libbinder",
-        "libcutils",
-        "libdl",
-        "libhidlbase",
-        "libutils",
-        "vendor.lineage.livedisplay@2.0",
-        "vendor.lineage.livedisplay@2.1",
-    ],
-    header_libs: [
-        "vendor.lineage.livedisplay@2.0-sdm-headers",
-    ],
-}
-
-cc_binary {
-    name: "vendor.lineage.livedisplay@2.1-service.eqe",
-    init_rc: ["vendor.lineage.livedisplay@2.1-service.eqe.rc"],
-    vintf_fragments: ["vendor.lineage.livedisplay@2.1-service.eqe.xml"],
-    defaults: ["livedisplay_eqe"],
-    vendor: true,
-}
diff --git a/livedisplay/AntiFlicker.cpp b/livedisplay/AntiFlicker.cpp
deleted file mode 100644
index 0fc5361..0000000
--- a/device/motorola/eqe/livedisplay/AntiFlicker.cpp
+++ /dev/null
@@ -1,58 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2019-2024 The LineageOS Project
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#include "AntiFlicker.h"
-
-#include <android-base/file.h>
-#include <android-base/strings.h>
-
-namespace {
-    constexpr const char *kFileDc = "/sys/devices/platform/soc/soc:qcom,dsi-display-primary/dsi_display_dc";
-};  // anonymous namespace
-
-using ::android::base::ReadFileToString;
-using ::android::base::Trim;
-using ::android::base::WriteStringToFile;
-
-namespace vendor {
-namespace lineage {
-namespace livedisplay {
-namespace V2_1 {
-namespace implementation {
-
-AntiFlicker::AntiFlicker() {
-    if (!access(kFileDc, R_OK | W_OK)) {
-        file_ = kFileDc;
-        enabled_mode_ = 1;
-    } else {
-        file_ = nullptr;
-    }
-}
-
-bool AntiFlicker::isSupported() {
-    return file_ != nullptr;
-}
-
-// Methods from ::vendor::lineage::livedisplay::V2_1::IAntiFlicker follow.
-Return<bool> AntiFlicker::isEnabled() {
-    std::string tmp;
-    int32_t contents = 0;
-
-    if (ReadFileToString(file_, &tmp)) {
-        contents = std::stoi(Trim(tmp));
-    }
-
-    return contents > 0;
-}
-
-Return<bool> AntiFlicker::setEnabled(bool enabled) {
-    return WriteStringToFile(enabled ? std::to_string(enabled_mode_) : "0", file_, true);
-}
-
-}  // namespace implementation
-}  // namespace V2_1
-}  // namespace livedisplay
-}  // namespace lineage
-}  // namespace vendor
diff --git a/livedisplay/AntiFlicker.h b/livedisplay/AntiFlicker.h
deleted file mode 100644
index 9f97514..0000000
--- a/device/motorola/eqe/livedisplay/AntiFlicker.h
+++ /dev/null
@@ -1,36 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2019-2024 The LineageOS Project
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#pragma once
-
-#include <vendor/lineage/livedisplay/2.1/IAntiFlicker.h>
-
-namespace vendor {
-namespace lineage {
-namespace livedisplay {
-namespace V2_1 {
-namespace implementation {
-
-using ::android::hardware::Return;
-
-class AntiFlicker : public IAntiFlicker {
-  public:
-    AntiFlicker();
-    bool isSupported();
-
-    // Methods from ::vendor::lineage::livedisplay::V2_1::IAntiFlicker follow.
-    Return<bool> isEnabled() override;
-    Return<bool> setEnabled(bool enabled) override;
-
-  private:
-    const char* file_;
-    int32_t enabled_mode_;
-};
-
-}  // namespace implementation
-}  // namespace V2_1
-}  // namespace livedisplay
-}  // namespace lineage
-}  // namespace vendor
diff --git a/livedisplay/SunlightEnhancement.cpp b/livedisplay/SunlightEnhancement.cpp
deleted file mode 100644
index 2f8e851..0000000
--- a/device/motorola/eqe/livedisplay/SunlightEnhancement.cpp
+++ /dev/null
@@ -1,58 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2019-2024 The LineageOS Project
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#include "SunlightEnhancement.h"
-
-#include <android-base/file.h>
-#include <android-base/strings.h>
-
-namespace {
-    constexpr const char *kFileHbm = "/sys/devices/platform/soc/soc:qcom,dsi-display-primary/dsi_display_hbm";
-};  // anonymous namespace
-
-using ::android::base::ReadFileToString;
-using ::android::base::Trim;
-using ::android::base::WriteStringToFile;
-
-namespace vendor {
-namespace lineage {
-namespace livedisplay {
-namespace V2_1 {
-namespace implementation {
-
-SunlightEnhancement::SunlightEnhancement() {
-    if (!access(kFileHbm, R_OK | W_OK)) {
-        file_ = kFileHbm;
-        enabled_mode_ = 1;
-    } else {
-        file_ = nullptr;
-    }
-}
-
-bool SunlightEnhancement::isSupported() {
-    return file_ != nullptr;
-}
-
-// Methods from ::vendor::lineage::livedisplay::V2_1::ISunlightEnhancement follow.
-Return<bool> SunlightEnhancement::isEnabled() {
-    std::string tmp;
-    int32_t contents = 0;
-
-    if (ReadFileToString(file_, &tmp)) {
-        contents = std::stoi(Trim(tmp));
-    }
-
-    return contents > 0;
-}
-
-Return<bool> SunlightEnhancement::setEnabled(bool enabled) {
-    return WriteStringToFile(enabled ? std::to_string(enabled_mode_) : "0", file_, true);
-}
-
-}  // namespace implementation
-}  // namespace V2_1
-}  // namespace livedisplay
-}  // namespace lineage
-}  // namespace vendor
diff --git a/livedisplay/SunlightEnhancement.h b/livedisplay/SunlightEnhancement.h
deleted file mode 100644
index 4248fad..0000000
--- a/device/motorola/eqe/livedisplay/SunlightEnhancement.h
+++ /dev/null
@@ -1,36 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2019-2024 The LineageOS Project
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#pragma once
-
-#include <vendor/lineage/livedisplay/2.1/ISunlightEnhancement.h>
-
-namespace vendor {
-namespace lineage {
-namespace livedisplay {
-namespace V2_1 {
-namespace implementation {
-
-using ::android::hardware::Return;
-
-class SunlightEnhancement : public ISunlightEnhancement {
-  public:
-    SunlightEnhancement();
-    bool isSupported();
-
-    // Methods from ::vendor::lineage::livedisplay::V2_1::ISunlightEnhancement follow.
-    Return<bool> isEnabled() override;
-    Return<bool> setEnabled(bool enabled) override;
-
-  private:
-    const char* file_;
-    int32_t enabled_mode_;
-};
-
-}  // namespace implementation
-}  // namespace V2_1
-}  // namespace livedisplay
-}  // namespace lineage
-}  // namespace vendor
diff --git a/livedisplay/service.cpp b/livedisplay/service.cpp
deleted file mode 100644
index 1e2c9cf..0000000
--- a/device/motorola/eqe/livedisplay/service.cpp
+++ /dev/null
@@ -1,80 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2019-2024 The LineageOS Project
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#define LOG_TAG "vendor.lineage.livedisplay@2.1-service.eqe"
-
-#include <android-base/logging.h>
-#include <binder/ProcessState.h>
-#include <hidl/HidlTransportSupport.h>
-#include <livedisplay/sdm/PictureAdjustment.h>
-#include <vendor/lineage/livedisplay/2.1/IPictureAdjustment.h>
-#include "AntiFlicker.h"
-#include "SunlightEnhancement.h"
-
-using ::android::OK;
-using ::android::sp;
-using ::android::status_t;
-using ::android::hardware::configureRpcThreadpool;
-using ::android::hardware::joinRpcThreadpool;
-
-using ::vendor::lineage::livedisplay::V2_0::sdm::PictureAdjustment;
-using ::vendor::lineage::livedisplay::V2_0::sdm::SDMController;
-using ::vendor::lineage::livedisplay::V2_1::IPictureAdjustment;
-using ::vendor::lineage::livedisplay::V2_1::implementation::AntiFlicker;
-using ::vendor::lineage::livedisplay::V2_1::implementation::SunlightEnhancement;
-
-status_t RegisterAsServices() {
-    status_t status = OK;
-
-    sp<SunlightEnhancement> se = new SunlightEnhancement();
-    if (se->isSupported()) {
-        status = se->registerAsService();
-        if (status != OK) {
-            LOG(ERROR) << "Could not register service for LiveDisplay HAL SunlightEnhancement Iface"
-                       << " (" << status << ")";
-            return status;
-        }
-    }
-
-	sp<AntiFlicker> af = new AntiFlicker();
-    if (af->isSupported()) {
-        status = af->registerAsService();
-        if (status != OK) {
-            LOG(ERROR) << "Could not register service for LiveDisplay HAL AntiFlicker Iface"
-                       << " (" << status << ")";
-            return status;
-        }
-    }
-
-    std::shared_ptr<SDMController> controller = std::make_shared<SDMController>();
-    sp<PictureAdjustment> pa = new PictureAdjustment(controller);
-	status = pa->registerAsService();
-    if (status != OK) {
-        LOG(ERROR) << "Could not register service for LiveDisplay HAL PictureAdjustment Iface ("
-                   << status << ")";
-        return status;
-    }
-
-    return OK;
-}
-
-int main() {
-	android::ProcessState::initWithDriver("/dev/vndbinder");
-
-    LOG(DEBUG) << "LiveDisplay HAL service is starting.";
-
-    configureRpcThreadpool(1, true /*callerWillJoin*/);
-
-    if (RegisterAsServices() == OK) {
-        LOG(DEBUG) << "LiveDisplay HAL service is ready.";
-        joinRpcThreadpool();
-    } else {
-        LOG(ERROR) << "Could not register service for LiveDisplay HAL";
-    }
-
-    // In normal operation, we don't expect the thread pool to shutdown
-    LOG(ERROR) << "LiveDisplay HAL service is shutting down.";
-    return 1;
-}
diff --git a/livedisplay/vendor.lineage.livedisplay@2.1-service.eqe.rc b/livedisplay/vendor.lineage.livedisplay@2.1-service.eqe.rc
deleted file mode 100644
index f258c30..0000000
--- a/device/motorola/eqe/livedisplay/vendor.lineage.livedisplay@2.1-service.eqe.rc
+++ /dev/null
@@ -1,11 +0,0 @@
-on init
-    # LiveDisplay sysfs
-    chown system system /sys/devices/platform/soc/soc:qcom,dsi-display-primary/dsi_display_hbm
-    chmod 0660 /sys/devices/platform/soc/soc:qcom,dsi-display-primary/dsi_display_hbm
-    chown system system /sys/devices/platform/soc/soc:qcom,dsi-display-primary/dsi_display_dc
-    chmod 0660 /sys/devices/platform/soc/soc:qcom,dsi-display-primary/dsi_display_dc
-
-service vendor.livedisplay-hal-2-1 /vendor/bin/hw/vendor.lineage.livedisplay@2.1-service.eqe
-    class late_start
-    user system
-    group system
diff --git a/livedisplay/vendor.lineage.livedisplay@2.1-service.eqe.xml b/livedisplay/vendor.lineage.livedisplay@2.1-service.eqe.xml
deleted file mode 100644
index 2cf6563..0000000
--- a/device/motorola/eqe/livedisplay/vendor.lineage.livedisplay@2.1-service.eqe.xml
+++ /dev/null
@@ -1,16 +0,0 @@
-<manifest version="2.0" type="device">
-	<hal format="hidl">
-        <name>vendor.lineage.livedisplay</name>
-        <transport>hwbinder</transport>
-        <version>2.1</version>
-        <interface>
-            <name>ISunlightEnhancement</name>
-            <instance>default</instance>
-        </interface>
-        <interface>
-            <name>IAntiFlicker</name>
-            <instance>default</instance>
-        </interface>
-        <fqname>@2.0::IPictureAdjustment/default</fqname>
-	</hal>
-</manifest>
diff --git a/device/motorola/eqe/overlay-lineage/packages/apps/Updater/app/src/main/res/values/strings.xml b/device/motorola/eqe/overlay-lineage/packages/apps/Updater/app/src/main/res/values/strings.xml
new file mode 100644
index 0000000..70d366b
--- /dev/null
+++ b/device/motorola/eqe/overlay-lineage/packages/apps/Updater/app/src/main/res/values/strings.xml
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <string
+        name="updater_server_url"
+        translatable="false"
+    >https://raw.githubusercontent.com/SomeEmptyBox/android_build_utils/refs/heads/main/ota/eqe/infinity.json</string>
+    <string
+        name="menu_changelog_url"
+        translatable="false"
+    >https://raw.githubusercontent.com/SomeEmptyBox/android_build_utils/refs/heads/main/ota/eqe/infinity.txt</string>
+</resources>
diff --git a/system.prop b/system.prop
index b968252..e19d63c 100644
--- a/device/motorola/eqe/system.prop
+++ b/device/motorola/eqe/system.prop
@@ -85,3 +85,10 @@ persist.sys.wfd.virtual=0
 
 #Disable MTE Async for system server
 arm64.memtag.process.system_server=off
+
+# InfinityX
+ro.product.marketname=Motorola Edge 50 Pro
+ro.infinity.soc=Snapdragon 7 Gen 3
+ro.infinity.battery=4500 mAh
+ro.infinity.display=1220 x 2712, 144 Hz
+ro.infinity.camera=Front 50MP / Rear 50MP + 10MP + 13MP
