diff --git a/device/motorola/eqe/device.mk b/device/motorola/eqe/device.mk
index 904890b..158b04b 100644
--- a/device/motorola/eqe/device.mk
+++ b/device/motorola/eqe/device.mk
@@ -341,12 +341,6 @@ PRODUCT_COPY_FILES += \
 PRODUCT_PACKAGES += \
     android.hardware.power-service-qti
 
-# PowerShare
-PRODUCT_PACKAGES += \
-    vendor.lineage.powershare-service.default
-
-$(call soong_config_set,lineage_powershare,powershare_path,/sys/class/power_supply/wireless/device/tx_mode)
-
 # Protobuf
 PRODUCT_PACKAGES += \
     libprotobuf-cpp-full-3.9.1-vendorcompat \
diff --git a/device/motorola/eqe/resource-overlay/eqe/Frameworks/res/values/config.xml b/device/motorola/eqe/resource-overlay/eqe/Frameworks/res/values/config.xml
index 0b1eb90..1f40032 100644
--- a/device/motorola/eqe/resource-overlay/eqe/Frameworks/res/values/config.xml
+++ b/device/motorola/eqe/resource-overlay/eqe/Frameworks/res/values/config.xml
@@ -266,7 +266,7 @@
     <!-- Not allow to switch to higher refresh rate when (display, ambient) brightness falls into
          the region defined by the two arrays -->
     <array name="config_brightnessThresholdsOfPeakRefreshRate">
-        <item>50</item>
+        <item>-1</item>
     </array>
     <array name="config_ambientThresholdsOfPeakRefreshRate">
         <item>-1</item>
diff --git a/device/motorola/eqe/device.mk b/device/motorola/eqe/device.mk
index 904890b..0f426c4 100644
--- a/device/motorola/eqe/device.mk
+++ b/device/motorola/eqe/device.mk
@@ -262,7 +262,6 @@ PRODUCT_PACKAGES += \
 # MotoActions
 PRODUCT_PACKAGES += \
     MotoActions \
-    MotoCommonOverlay
 
 # NFC
 PRODUCT_PACKAGES += \
